.pipelines/.vsts-garabge-collection.yaml

```diff
@@ -0,0 +1,20 @@
+pool:
+  name: $(POOL_NAME)
+
+parameters:
+- name: DRY_RUN
+  displayName: Dry Run
+  type: boolean
+  default: false
+
+jobs:
+- job: gc
+  displayName: Garbage Collection
+  steps:
+    - bash: |
+        chmod +x ./vhdbuilder/rg-cleanup.sh
+        ./vhdbuilder/rg-cleanup.sh
+      env:
```

Comment:
we need to have a TAG that the GC ignores.  when debugging Build issue or e2e issues it might be best to leave some clusters/vmss online. 

we could look at force deleting after 7 days

Reply:
added

Reply:
We had it for e2e and I didn't like. People often use for debugging and often don't clean after themselves.
So, the resources are wasted for a long time.

I like the simplicity of deleting everything overnight. No complicated logic, no opportunity to leave long-leaving resources, completely predictable.

A day should be enough to debug things. Next day things can be re-created again.

Reply:
We probably need to have one for some edge-cases and long-living resources. But I'd like to avoid using such tags unless it's absolutely necessary. I don't want to have an option to extend the lifetime. I'd rather everything to either live forever or a single day.

Reply:
I think the current strategy is okay since "by default", things will be removed within 24 hours of creation

you'd need to explicitly add the `gc.skip=true` to the resource to have it stay around longer, and even then it will still be forcibly removed after 7 days

Reply:
we can also iterate on this if we find that other behavior works better in the future